dev/Repeater/APITests/ViewManagerTests.cs

```diff
@@ -759,6 +759,76 @@ public void ValidateFocusMoveOnElementClearedWithUniqueIds()
+        // To check if the clearing is working correctly, we log all data contexts, and check if all of them are null.
+        public void ValidateElementClearingClearsDataContext()
+        {
+            ItemsRepeater repeater = null;
+            MockRecyclingTestFactory elementFactory = null;
+            int elementClearingRaisedCount = 0;
+            ItemsRepeaterScrollHost scroller = null;
+            Log.Comment("Initialize ItemsRepeater");
+            RunOnUIThread.Execute(() =>
+            {
+                elementFactory = new MockRecyclingTestFactory();
+                elementFactory.RecyclePool = new RecyclePool();
+                elementFactory.Templates["Item"] = (DataTemplate)XamlReader.Load(
+                    @"<DataTemplate xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'> " + "<Button Content=\"{Binding}\" />" + @"</DataTemplate>");
+
+                repeater = CreateRepeater(Enumerable.Range(0, 100),
+                    elementFactory);
+
+                repeater.Layout = new StackLayout();
+
+                var scrollViewer = new ScrollViewer() { Content = repeater };
+                scroller = new ItemsRepeaterScrollHost() 
+                {
+                    Height = 200,
+                    Width = 200,
+                    ScrollViewer = scrollViewer
+                };
+
+                Content = scroller;
+                scroller.UpdateLayout();
+                repeater.ElementClearing += Repeater_ElementClearing;
+            });
+
+            IdleSynchronizer.Wait();
+
+            Log.Comment("Scrolling ItemsRepeater around to clear and recycle elements");
+            RunOnUIThread.Execute(() =>
+            {
+                scroller.ScrollViewer.ChangeView(null, 400, null, true);
+                scroller.ScrollViewer.ChangeView(null, 600, null, true);
+                scroller.ScrollViewer.ChangeView(null, 800, null, true);
+            });
+
+            IdleSynchronizer.Wait();
+
+            Log.Comment("Verify ItemsRepeater cleared data contexts correctly");
+            Verify.IsTrue(elementClearingRaisedCount > 0, "ItemsRepeater should have cleared some elements");
+            foreach (object o in elementFactory.DataContextsInGetElement)
+            {
+                // TL;DR; on why we check this:
+                // All DataContexts that are present on our returned ItemTemplate(Shim) get logged inside DataContextsInGetElement.
+                // If one of them is not null, that means the DataContext did not get cleared correctly which is a bug.
+                Verify.IsNull(o, "Datacontext should be null on a recycled ItemTemplate");
+            }
+
+            void Repeater_ElementClearing(ItemsRepeater sender, Microsoft.UI.Xaml.Controls.ItemsRepeaterElementClearingEventArgs args)
+            {
+                elementClearingRaisedCount++;
+            }
```

Comment:
can do this increment in ClearElementFunc also to verify that it got called atleast once.

Reply:
Done.